core/payment/src/api/allocations.rs

```diff
@@ -46,44 +112,147 @@ async fn create_allocation(
     // TODO: Handle deposits & timeouts
     let allocation = body.into_inner();
     let node_id = id.identity;
+
+    const DEFAULT_PAYMENT_PLATFORM: &str = "erc20-holesky-tglm";
+    const DEFAULT_PAYMENT_PLATFORM_FOR_TGLM: &str = "erc20-holesky-tglm";
+    const DEFAULT_PAYMENT_PLATFORM_FOR_GLM: &str = "erc20-polygon-glm";
+
     let payment_platform = match &allocation.payment_platform {
-        Some(platform) => platform.clone(),
-        None => return response::bad_request(&"payment platform must be provided"),
+        Some(PaymentPlatformEnum::PaymentPlatformName(name)) => {
+            log::debug!("Using old API payment platform name as pure str: {}", name);
+            name.clone()
+        }
+        Some(PaymentPlatformEnum::PaymentPlatform(p)) => {
+            if p.driver.is_none() && p.network.is_none() && p.token.is_none() {
+                log::debug!("Empty paymentPlatform object, using {DEFAULT_PAYMENT_PLATFORM}");
+                DEFAULT_PAYMENT_PLATFORM.to_string()
+            } else if p.token.is_some() && p.network.is_none() && p.driver.is_none() {
+                let token = p.token.as_ref().unwrap();
+                if token == "glm" {
+                    log::debug!("Selected network {DEFAULT_PAYMENT_PLATFORM_FOR_GLM} (default for glm token)");
+                    DEFAULT_PAYMENT_PLATFORM_FOR_GLM.to_string()
+                } else if token == "tglm" {
+                    log::debug!("Selected network {DEFAULT_PAYMENT_PLATFORM_FOR_TGLM} (default for tglm token)");
+                    DEFAULT_PAYMENT_PLATFORM_FOR_TGLM.to_string()
+                } else {
+                    let err_msg =
+                        format!("Only glm or tglm token values are accepted vs {token} provided");
+                    return response::bad_request(&err_msg);
+                }
+            } else {
+                let network = p.network.as_deref().unwrap_or_else(|| {
+                    if let Some(token) = p.token.as_ref() {
+                        if token == "glm" {
+                            log::debug!("Network not specified, using default polygon, because token set to glm");
+                            "polygon"
+                        } else {
+                            log::debug!("Network not specified, using default holesky");
+                            "holesky"
+                        }
+                    } else {
+                        log::debug!("Network not specified and token not specified, using default holesky");
+                        "holesky"
+                    }
```

Comment:
Driver has `get_default_network` method. Probably you should use it, otherwise defaults will diverge.